city_scrapers/spiders/il_procurement_policy.py

```diff
@@ -0,0 +1,153 @@
+        if not len(title_str):
+            return ""
+        name_str = title_str[len(title_str) - 1] + " Board Meeting Minutes"
+        return name_str
+
+    def _parse_description(self, item):
+        """Parse or generate meeting description."""
+        return ""
+
+    def _parse_classification(self, item):
+        """Parse or generate classification from allowed options."""
+        return BOARD
+
+    def _parse_start(self, item):
+        """Parse start datetime as a naive datetime object."""
+        time_object = time(10, 0)
+        date_str = " ".join(item.css("*::text").extract()).strip()
+        date_str = re.sub("Agenda.pdf", "", date_str).strip()
+        date_object = datetime.strptime(date_str, "%B %d, %Y").date()
+        return datetime.combine(date_object, time_object)
+
+    def _parse_links(self, item, response):
+        """Parse or generate links."""
+        links = []
+        title_str = " ".join(item.css("*::text").extract()).strip()
+        if 'pdf' in title_str:
+            title_str = re.sub("Agenda.pdf", "", title_str).strip()
+            title_str += " Agenda"
+        links.append({
+            'title': title_str,
+            'href': response.urljoin(item.attrib['href']),
+        })
+        return links
+
+    def _parse_past_links(self, response):
+        """parse start datetime as a naive datetime object"""
+        links = []
+        index = None
+        for item in response.css(".ms-rtestate-field p a"):
+            title_str = " ".join(item.css("*::text").extract()).strip()
+            title_str = title_str.replace("\u200b", "")
+            if '.pdf' in title_str:
+                index = title_str.index('.pdf')
+            if '[' in title_str:
+                index = title_str.index('[')
+            if '-' in title_str:
+                index = title_str.index('-')
+            title_str = title_str[:index]
+            links.append({'title': title_str, 'href': response.urljoin(item.attrib['href'])})
+        for item in response.css(".ms-rtestate-field .list-unstyled li a"):
+            title_str = " ".join(item.css("*::text").extract()).strip()
+            title_str = re.sub(".pdf", "", title_str).strip()
+            title_str = title_str.replace("\u200b", "")
+            index = len(title_str)
+            if '- Amended' in title_str:
+                title_str = title_str.replace("- Amended", "").strip()
+            if '[' in title_str:
+                index = title_str.index('[')
+            if '-' in title_str:
+                index = title_str.index('-')
+            title_str = title_str[:index].strip()
+            links.append({'title': title_str, 'href': response.urljoin(item.attrib['href'])})
+        return links
+
+    def _past_start(self, item):
+        """parse or generate links from past meetings"""
+        time_object = time(10, 0)
+        date_str = item['title']
+        if not len(date_str):
+            date_str = "December 2, 2015"
```

Comment:
Where is this date coming from?

Reply:
There were two pdfs given for that date but no text or date given for one of the pdfs.

Reply:
I’d rather ignore that one than hard code this since it’s likely it’ll change